tests/test_modes.py

```diff
@@ -173,3 +175,75 @@ def test_should_only_list_pytest_root_changed_files(self, git_repository, testdi
         pytestroot.chdir()
         output = set(Branch([]).git_output().splitlines())
         assert output == {"A       test_pytestroot.py"}
+
+
+class TestOnlyChanged:
+    def test_should_return_git_diff_command(self):
+        mode = OnlyChanged([])
+        command = mode.command()
+
+        assert isinstance(command, list)
+        assert mode.command() == ["git", "diff"]
+
+    @pytest.mark.parametrize(
+        "line,expected_line",
+        [
+            (
+                "@@ -191,29 +191,8 @@ class TestPytestPicked:",
+                "@@ -191,29 +191,8 @@ class TestPytestPicked:",
+            ),
+            ("-    def test_positive_update_limit(self):", None),
+            ("--- a/tests/foreman/cli/test_host_collection.py", None),
+            (
+                "+++ b/tests/pytestpicked/test_positive",
+                "tests/pytestpicked/test_positive",
+            ),
+            ("+    @pytest.mark.tier2", "+    @pytest.mark.tier2"),
+            ("+    def test", "+    def test"),
+            ("-def test_positive", None),
+        ],
+    )
+    def test_parser_should_ignore_no_paths_characters(self, line, expected_line):
+        mode = OnlyChanged([])
+        parsed_line = mode.parser(line)
+
+        assert parsed_line == expected_line
+
+    def test_should_list_changed_tests(self):
+        raw_output = (
+            b"\n+++ b/tests/pytestpicked/test_modes1.py"
+            b"\n@@ -191,29 +191,8 @@ class TestPytestPicked(CLITestCase):"
+            b"\n+    def test_with_class(self):"
+            b"\n+++ b/tests/pytestpicked/test_modes1.py"
+            b"\n@@ -191,29 +191,8 @@ class TestPytestPicked(CLITestCase):"
+            b"\n+++ b/tests/pytestpicked/test_modes1.py"
+            b"\n+    def test_with_duplicate_classes(self):"
+            b"\n+++ b/tests/pytestpicked/test_modes2.py"
+            b"\n@@ -191,29 +191,8 @@ class TestPytestPicked:"
+            b"\n+    def test_with_class_1(self):"
+            b"\n+    def test_with_class_2(self):"
+            b"\n+++ b/tests/pytestpicked/test_modes3.py"
+            b"\n+    def test_without_class(self):"
+            b"\n+++ b/tests/pytestpicked/test_modes4.py"
+            b"\n+    def invalid_function(self):"
+            b"\n+++ b/tests/pytestpicked/test_modes5.py"
+            b"\n+    def invalid_test(self):"
+            b"\n+++ b/tests/pytestpicked/test_modes6.py"
+            b"\n@@ -191,29 +191,8 @@ class InvalidClass:"
```

Comment:
All lines are about the things we want to extract. But it isn't how real diffs looks like. So would be nice to have some real diff here. Feel free to add a file in a directory in our test folder if you think it would be better.